cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/QueryStates.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.query
+
+import de.fraunhofer.aisec.cpg.assumptions.AssumptionStatus
+
+typealias Decision = QueryTree<DecisionState>
+
+sealed class DecisionState
+
+/**
+ * Represents a query that has been evaluated and the result is known to be `false` or some
+ * [QueryTree.assumptions] have [de.fraunhofer.aisec.cpg.assumptions.Assumption.status]
+ * [AssumptionStatus.Rejected].
+ */
+data object Failed : DecisionState()
+
+/**
+ * Represents a query that has been evaluated and the result is known to be `true` and all
+ * [QueryTree.assumptions] have [de.fraunhofer.aisec.cpg.assumptions.Assumption.status]
+ * [AssumptionStatus.Accepted] or [AssumptionStatus.Ignored].
+ */
+data object Succeeded : DecisionState()
+
+/**
+ * Represents a query that has been evaluated but the result is not yet known because some
+ * [QueryTree.assumptions] have [de.fraunhofer.aisec.cpg.assumptions.Assumption.status]
+ * [AssumptionStatus.Undecided].
+ */
+data object Undecided : DecisionState()
+
+/**
+ * Represents a query that has not yet been evaluated. Will be most likely used in the context of
+ * manual assessments which have to be conducted.
+ */
+data object NotYetEvaluated : DecisionState()
+
+fun QueryTree<Boolean>.decide(): Decision {
```

Comment:
Since this will a public facing API, please generate a short docstring for this.